### PR TITLE
Remove deprecated Geo functions

### LINKED ISSUE #283: Remove deprecated geocode functions
The refactor #217 has deprecated a number of functions. They haven't been removed so as not to break the recent usage of the utility in JHU, Quidel, and possibly other indicators. The functions are of the form `geocode_to_geocode` or `convert_geocode_to_geocode`, e.g. `zip_to_msa` and `convert_zip_to_msa`.

### PR DESCRIPTION
**Pull request by chinandrew**
### Description
With the JHU and Quidel georefactors complete, these are no longer used.

### Changelog
Itemize code/test/documentation changes and files added/removed.
- Remove TODO comment 
- Remove deprecated functoins

### Fixes 
- Fixes #283

### CODE REVIEW

**Review by sgsmob — APPROVED** [review]

**Comment by chinandrew** [comment]
oof the tests use both and compare against each other, which is a bit annoying. I removed all the tests for old functions and the ones that compared the old to new ones.

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 1]
```diff
@@ -6,15 +6,12 @@
...
-- remove deprecated functions once integration into JHU and Quidel is refactored
-  see: https://github.com/cmu-delphi/covidcast-indicators/issues/283
 """
 # pylint: disable=too-many-lines
 from os.path import join
-import warnings
-import pkg_resources
 
 import pandas as pd
+import pkg_resources
```
Comment (sgsmob) [inline _delphi_utils_python/delphi_utils/geomap.py:14]: Why is this needed?

**Review by sgsmob — APPROVED** [review]

File: _delphi_utils_python/delphi_utils/geomap.py (hunk 1 above)
Comment (chinandrew) [reply _delphi_utils_python/delphi_utils/geomap.py:14]: it just got auto-reordered by pycharm when I was cleaning up imports, used to be line 15.
